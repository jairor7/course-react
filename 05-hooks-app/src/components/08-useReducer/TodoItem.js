import React from "react";
import PropTypes from "prop-types";

const TodoItem = ({
  todo: t,
  index,
  handleComplete,
  handleRemoveTodo,
}) => {
  return (
    <li key={t.id} className="list-group-item">
      <p
        className={`todo-task${t.done ? " complete" : ""}`}
        onClick={() => handleComplete(t.id)}
      >
        {index + 1}. {t.description}
      </p>
      <button
        className="btn btn-danger"
        onClick={() => handleRemoveTodo(t.id)}
      >
        Borrar
      </button>
    </li>
  );
};

TodoItem.propTypes = {
  todo: PropTypes.object.isRequired,
  // index: PropTypes.number.isRequired,
  // handleComplete: PropTypes.func.isRequired,
  // handleRemoveTodo: PropTypes.func.isRequired,
};

export default TodoItem;
