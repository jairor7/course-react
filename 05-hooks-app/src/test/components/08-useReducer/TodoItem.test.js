import TodoItem from "../../../components/08-useReducer/TodoItem";
import { shallow } from "enzyme";
import { demoInitialStateTodos } from "../../fixtures/demoTodos";

describe("Pruebas en <TodoItem />", () => {
  const handleComplete = jest.fn();
  const handleRemoveTodo = jest.fn();
  const wrapper = shallow(
    <TodoItem
      todo={demoInitialStateTodos[0]}
      index={demoInitialStateTodos[0].id}
      handleComplete={handleComplete}
      handleRemoveTodo={handleRemoveTodo}
    />
  );
  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de llamar la funcion handleRemoveTodo", () => {
    wrapper.find("button").simulate("click");
    expect(handleRemoveTodo).toHaveBeenCalledWith(
      demoInitialStateTodos[0].id
    );
  });

  test("Debe de llamar la funcion handleComplete", () => {
    wrapper.find("p").simulate("click");
    expect(handleComplete).toHaveBeenCalledWith(
      demoInitialStateTodos[0].id
    );
  });

  test("Debe de mostrar el texto correctamente del parrafo", () => {
    expect(wrapper.find("p").text()).toBe(
      `${demoInitialStateTodos[0].id + 1}. ${
        demoInitialStateTodos[0].description
      }`
    );
  });

  test("Debe el parrafo tener la clase complete, si la tarea esta en done", () => {
    const todo = demoInitialStateTodos[0];
    todo.done = true;
    const wrapper = shallow(<TodoItem todo={todo} />);
    // const { className } = wrapper.find("p").props();
    // expect(className.includes("complete")).toBe(true);
    expect(wrapper.find("p").hasClass("complete")).toBe(
      true
    );
  });
});
