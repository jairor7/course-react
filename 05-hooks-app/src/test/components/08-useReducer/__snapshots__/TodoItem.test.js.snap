// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoItem /> Debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="todo-task"
    onClick={[Function]}
  >
    2
    . 
    Hacer la compra
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
